### PR TITLE
API-27098: Updates to Handling of Duplicate UUID Response from Central Mail (Appeals and Benefits Intake)

### PR DESCRIPTION
**Pull request by kristen-brown**
## Summary
Some recent observations after changing our error handling slightly in the Appeals API have led us to the understanding that a status code of `400` with a response body containing "Document already uploaded with uuid," when returned from the Central Mail `/upload` endpoint, indicates that there was an earlier submission (with that same UUID) that was successful, and so the submission record itself can be marked as successful. This new understanding has led to the changes in this PR, which counts "duplicate UUID" responses as always successful in both the Appeals API and Benefits Intake API and adds a StatsD counter and Rails logs for the "duplicate UUID" scenario so that we can more easily monitor how often it occurs.

## Related issue(s)
[API-27098](https://jira.devops.va.gov/browse/API-27098)

## Testing done
The `AppealsApi::PdfSubmitJob` and `VBADocuments::UploadProcessor` are well-covered by existing unit tests. I have also added unit tests specifically for the "duplicate UUID" scenario for both classes.

## Screenshots
None

## What areas of the site does it impact?
This PR impacts the Appeals API and the Benefits Intake API, both of which my team own.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation) – N/A
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable) – TBD
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected – N/A
- [ ]  I added a screenshot of the developed feature – N/A

## Requested Feedback
No specific feedback requested on this PR.

### CODE REVIEW

File: modules/appeals_api/app/workers/appeals_api/pdf_submit_job.rb [hunk 1]
```diff
@@ -30,6 +30,8 @@ class PdfSubmitJob
     # 503 Database Offline || SOLR Service Offline || Intake API is undergoing maintenance
     RETRYABLE_EMMS_RESP_STATUS_CODES = [401, 429, 500, 503].freeze
 
+    STATSD_DUPLICATE_UUID_KEY = 'api.appeals.document_upload.duplicate_uuid'
```
Comment (kristen-brown) [inline modules/appeals_api/app/workers/appeals_api/pdf_submit_job.rb:33]: The `STATSD_UPLOAD_FAIL_KEY` for Appeals is set to `api.appeals.document_upload.fail`, so I was trying to keep things consistent here.

**Review by mathisto — APPROVED** [review]
seal-of-approval-thumbs-up-gif

File: modules/vba_documents/spec/workers/upload_processor_spec.rb [hunk 2]
```diff
@@ -258,6 +258,36 @@
       end
     end
 
+    context 'with a duplicate UUID response from Central Mail' do
+      before do
```
Comment (mathisto) [inline modules/vba_documents/spec/workers/upload_processor_spec.rb:262]: I'm legit intimidated by your rspec-fu
